src/background/legacy-external-message-handler.ts

```diff
@@ -105,7 +107,15 @@ export async function handleLegacyExternalMethodFormat(
     case ExternalMethods.transactionRequest: {
       const { urlParams, tabId } = makeSearchParamsWithDefaults(port, [['request', payload]]);
 
-      const { id } = await triggerRequstWindowOpen(RouteUrls.TransactionRequest, urlParams);
+      const { network } = getPayloadFromToken(payload);
+
+      if (network) {
+        const developerDefinedApiUrl = getCoreApiUrl(network);
+        urlParams.set('coreApiUrl', developerDefinedApiUrl);
+        urlParams.set('networkChainId', network.chainId.toString());
+      }
```

Comment:
Could we pass these in the same way we do request payload?

Reply:
Could make a function to avoid this repetition below

```ts
function getNetworkParamsFromPayload(payload: string) {
  const { network } = getPayloadFromToken(payload);

  if (!network) return [];
  const developerDefinedApiUrl = getCoreApiUrl(network);
  return [
    ['coreApiUrl', developerDefinedApiUrl],
    ['networkChainId', network.chainId.toString()]
  ];
}

const { urlParams, tabId } = makeSearchParamsWithDefaults(port, [
  ['request', payload], 
  ...getNetworkParamsFromPayload(payload)
]);
```

Reply:
I don't remember resolving this. I'll fix, sorry.